prow/pod-utils/gcs/upload.go

```diff
@@ -171,12 +197,51 @@ func (w gcsObjectWriter) ApplyAttributes(attrs *storage.ObjectAttrs) {
 	w.Writer.ObjectAttrs = *attrs
 }
 
+type openerObjectWriter struct {
+	pkgio.Opener
+	Bucket      string
+	Dest        string
+	writeCloser pkgio.WriteCloser
+}
+
+func (w *openerObjectWriter) Write(p []byte) (n int, err error) {
+	if w.writeCloser == nil {
+		w.writeCloser, err = w.Opener.Writer(context.Background(), fmt.Sprintf("%s/%s", w.Bucket, w.Dest))
```

Comment:
The context should be part of the `openerObjectWriter`

Reply:
if we ever want to use this for GCS as well or pass any kind of opts, we need to change the `Writer` method in the `io` package to accept some kind of opts. Could probably be a simplistic struct that only contains GCS options and gets extended in the future if the need arises. We can do this backwards-compatible by leveraging the functional opts pattern.

Reply:
Let's keep this comment about the context and the other ones about the options. context is fixed